File: frontend/src/typings/api/get-team-leaderboard-response.ts
/* tslint:disable */
import { TeamLeaderboardEntryDto } from './team-leaderboard-entry-dto';

export interface GetTeamLeaderboardResponse {
  leaderboard: Array<TeamLeaderboardEntryDto>;
  page_count: number;
  user_team_ranking?: TeamLeaderboardEntryDto;
}

File: api/src/models/http/responses/get_team_leaderboard_response.rs
use crate::models::team::TeamLeaderboardEntryDto;

#[derive(serde::Serialize, serde::Deserialize, rocket_okapi::JsonSchema)]
pub struct GetTeamLeaderboardResponse {
    pub leaderboard: Vec<TeamLeaderboardEntryDto>,
    pub page_count: usize,
    pub user_team_ranking: Option<TeamLeaderboardEntryDto>,
}
